README.md: Add Linux build instructions

MordiJC opened the pull request:
Adding simple building instructions for Linux users. Some dependencies might be missing, because it took me some time to get them, but it's better than nothing :D

MordiJC committed:
README.md: Add Linux build instructions

MordiJC commented:
Wait, do not merge yet. I have to fix one thing

MordiJC commented:
Ok, now instructions are mostly in line with GitHub workflow. It should be easier to keep it up-to-date.

sthalik commented:
It's missing information on the optional modules, which is what the majority of people want (opencv, onnxruntime). `SDK_XPLANE` already defaults to an empty string. `ninja install` does the moral equivalent of `ninja -j$(nproc) && ninja install` and is shorter.

MordiJC commented:
> It's missing information on the optional modules, which is what the majority of people want (opencv, onnxruntime). 

Done. I added what I could find. Please let me know if it's enough.
I omitted Razer Hydra because I couldn't find any way to obtain Sixense SDK.

> `SDK_XPLANE` already defaults to an empty string.

True. I was using `cmake-gui` to inspect variables and it must have set it to non-empty, so that's why I added it to instructions. I'll remove it then.

> `ninja install` does the moral equivalent of `ninja -j$(nproc) && ninja install` and is shorter.

Fixed.

MordiJC commented:
> I'd like to put it on the wiki and hyperlink to it from the readme file. 

Sure, it seems reasonable. I'd leave essential building instructions in the README.md and have more exhaustive instructions on the wiki / in docs.

Also, I suggest archiving wiki somewhere in case GitHub decided to drop wiki support or do some other crazy "streamlining". ([GitHub Wiki How-To ](https://gist.github.com/subfuzion/0d3f19c4f780a7d75ba2))

decorahensisire commented:
Opentrack support for Linux is a lie.

DaWelter commented on README.md:
```diff
@@ -103,3 +103,106 @@ The code is held to a high-quality standard and written with utmost care; consid
+
+Dependencies: `libopencv-dev`, [onxxruntime tarball](https://github.com/microsoft/onnxruntime/releases)
+
+Note: Set `ONXXRuntime_ROOT` environment variable to path of downloaded and extracted onxxruntime tarball.
+Set `ONXXRuntime_DIR` to same value.
```
Nit:
```suggestion
Note: Set `ONXXRuntime_DIR` environment variable to path of downloaded and extracted onxxruntime tarball.
```
I wasn't aware `ONXXRuntime_ROOT` existed. It should be fine like this.

tartley commented:
Be aware, there is already a page in the opentrack wiki that describes [building on Linux](https://github.com/tartley/opentrack-wiki/blob/master/Building-on-Linux.md). If someone wants to merge this PR, I could happily submit a PR merging the two, in whichever location you prefer...

tartley commented:
I'm working on merging this into the wiki page now. This is a fabulous piece of work by @MordiJC. As a stranger to the project it has helped me immensely. Thank you @MordiJC!
